src/main/kotlin/io/github/anki/anki/service/DeckService.kt

```diff
@@ -1,26 +1,63 @@
 package io.github.anki.anki.service
 
+import io.github.anki.anki.repository.mongodb.CardRepository
 import io.github.anki.anki.repository.mongodb.DeckRepository
+import io.github.anki.anki.repository.mongodb.document.MongoDeck
+import io.github.anki.anki.service.exceptions.DeckDoesNotExistException
 import io.github.anki.anki.service.model.Deck
 import io.github.anki.anki.service.model.mapper.toDeck
 import io.github.anki.anki.service.model.mapper.toMongo
-import org.slf4j.LoggerFactory
+import org.bson.types.ObjectId
 import org.springframework.stereotype.Service
 
 @Service
 class DeckService(
     private val deckRepository: DeckRepository,
+    private val cardRepository: CardRepository,
 ) {
     fun createNewDeck(deck: Deck): Deck {
-        LOG.info("Creating new deck: {}", deck)
-        return deckRepository.insert(
-            deck.toMongo(),
-        )
+        return deckRepository
+            .insert(deck.toMongo())
+            .toDeck()
+    }
+
+    fun getDecks(userId: String): List<Deck> {
+        return deckRepository
+            .findByUserIdWithStatus(ObjectId(userId))
+            .map { it.toDeck() }
+    }
+
+    fun updateDeck(deck: Deck): Deck {
+        val mongoDeck =
+            getDeckByIdAndUserId(
+                deck.id ?: throw IllegalArgumentException("Deck id can not be null"),
+                deck.userId,
+            )
+        val updatedMongoDeck = mongoDeck.update(deck)
+        if (mongoDeck == updatedMongoDeck) {
+            return mongoDeck.toDeck()
+        }
+        return deckRepository
+            .save(updatedMongoDeck)
             .toDeck()
-            .also { LOG.info("Successfully saved new collection: {}", it) }
     }
 
-    companion object {
-        private val LOG = LoggerFactory.getLogger(CardsService::class.java)
+    fun deleteDeck(deckId: String) {
+        deckRepository.softDelete(ObjectId(deckId))
+        cardRepository.softDeleteByDeckId(ObjectId(deckId))
+    }
+
+    fun getDeckByIdAndUserId(deckId: String, userId: String): MongoDeck {
```

Comment:
Method looks weird. You don't need userId to get deck. The deck enough. But if you want to check user permissions, then I suggest to rename method and return boolean

```kotlin
fun hasPermissions(userId: String, deckId: String): Boolean {
        return deckRepository.findByIdAndUserIdWithStatus(ObjectId(deckId), ObjectId(userId)) != null
    }
```